scripts/deployment/deploy_locally.py

```
@@ -350,9 +350,32 @@
+  *) CONSTRAINTS_PATH="$PROJECT_ROOT/$CONSTRAINTS_PATH" ;;
+esac
+
 # Install dependencies
 echo "📦 Installing dependencies..."
-pip install -r requirements.txt
+if [ -f "$CONSTRAINTS_PATH" ]; then
+  echo "🔒 Using constraints file: $CONSTRAINTS_PATH"
+  pip install -c "$CONSTRAINTS_PATH" -r requirements.txt
+else
+  echo "⚠️ Constraints file not found at: $CONSTRAINTS_PATH"
+  echo "ℹ️ Proceeding without constraints (consider providing a valid path as first arg or CONSTRAINTS_PATH env var)"
```

Ruff (0.12.2)

String contains ambiguous `ℹ` (INFORMATION SOURCE). Did you mean `i` (LATIN SMALL LETTER I)?

(RUF001)